babs/cli.py

```diff
@@ -1138,13 +1138,68 @@ def check_df_job_specific(df, job_status_path_abs, type_session, which_function)
     return df
 
 
+def _parse_sync_code():
+    """Create and configure the argument parser for the `babs sync-code` command.
+
+    Returns
+    -------
+    argparse.ArgumentParser
+    """
+    parser = argparse.ArgumentParser(
+        description='Save and push code changes to input dataset.',
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument(
+        'project_root',
+        nargs='?',
+        default=os.getcwd(),
+        help=(
+            'Absolute path to the root of BABS project. '
+            "For example, '/path/to/my_BABS_project/' "
+            '(default is current working directory).'
+        ),
+    )
+    parser.add_argument(
+        '-m',
+        '--message',
+        help='Commit message for datalad save',
+        default='[babs] sync code changes',
+    )
+
+    return parser
+
+
+def babs_sync_code_main(project_root: str, commit_message: str):
+    """This is the core function of babs sync-code.
+
+    Parameters
+    ----------
+    project_root: str
+        absolute path to the directory of BABS project
+    commit_message: str
+        commit message for datalad save
+    """
+    # Get class `BABS` based on saved `analysis/code/babs_proj_config.yaml`:
+    babs_proj, _ = get_existing_babs_proj(project_root)
+
+    # Change to analysis directory
+    analysis_dir = os.path.join(project_root, 'analysis')
+    if not os.path.exists(analysis_dir):
+        raise Exception(f'Analysis directory does not exist at: {analysis_dir}')
```

Comment:
```suggestion
        raise FileNotFoundError(f'Analysis directory does not exist at: {analysis_dir}')
```

Apparently in Python they use FileNotFoundError for nonexistent folders too. E.g., try `import os;os.listdir("folder_that_doesnt_exist/")`